src/functional/ffront/type_info.py

```diff
@@ -528,13 +558,30 @@ def function_signature_incompatibilities_scanop(
         yield f"Scan operator takes {len(scanop_type.definition.args)-1} arguments, but {len(args)} were given."
         return
 
+    promoted_args = []
+    for i, scan_pass_arg in enumerate(scanop_type.definition.args[1:]):
+        # Helper function that given a scalar type in the signature of the scan
+        # pass return a field type with that dtype and the dimensions of the
+        # corresponding field type in the requested `args` type. Defined here
+        # as we capture `i`.
+        def _as_field(dtype: ct.ScalarType, path: tuple[int, ...]):
```

Comment:
I don't understand why the function definition is inside the for loop

Reply:
I tried to explain that in the comment. The function uses the iteration variable `i` inside. I could write a more complicated function outside, but this felt easier to me. I am open to suggestions, as I don't particularly like it either.